libs/ngxtension/lazy-service/src/lazy-service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injector, Type, runInInjectionContext } from '@angular/core';
+import { assertInjector } from 'ngxtension/assert-injector';
+import { Observable, defer } from 'rxjs';
+
+/**
+ * Loads a service lazily. The service is loaded when the observable is subscribed to.
+ *
+ * @param loader A function that returns a promise of the service to load.
+ * @param injector The injector to use to load the service. If not provided, the current injector is used.
+ * @returns An observable of the service.
+ *
+ * @example
+ * ```ts
+ * const dataService$ = lazyService(() => import('./data-service').then((m) => m.MyService));
+ * or
+ * const dataService$ = lazyService(() => import('./data-service'));
+ * ```
+ */
+export function lazyService<T>(
+	loader: () => Promise<Type<T>> | Promise<{ default: Type<T> }>,
+	injector?: Injector
+): Observable<T> {
+	injector = assertInjector(lazyService, injector);
+
+	return runInInjectionContext(injector, () => {
+		return defer(() => {
+			return loader()
```

Comment:
While this function works fine, it kind of bothers me that we can't mock the loaded instance. 

See http://riegler.fr/blog/2023-09-30-lazy-loading-mockable For an implementation that provides suck mocking abilities !